crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -224,28 +226,67 @@ where
         &mut self,
         input: &Message,
     ) -> Result<Vec<Message>, ConversionError> {
-        let tedge_event = ThinEdgeEvent::try_from(input.topic.name.as_str(), input.payload_str()?)?;
-        let c8y_event = C8yCreateEvent::try_from(tedge_event)?;
+        let mut vec = Vec::new();
+        let c8y_event;
+        // check if there is a childid in the topic, if not create the child before forwarding the event message
+        let child_id = get_child_id_from_event_topic(&input.topic.name)?;
```

Comment:
`child_id` can be `None`.
```suggestion
        let maybe_child_id = get_child_id_from_event_topic(&input.topic.name)?;
```

Reply:
This function does not exists any more.